packages/ui-kit/lib/components/video-player.tsx

```diff
@@ -0,0 +1,164 @@
+        }
+        
+        return () => {
+            if (timeoutRef.current) {
+                clearTimeout(timeoutRef.current);
+            }
+        };
+    }, [videoId, thumbnailUrl, dictionary, onErrorCallback, loadTimer]);
+
+    const handleThumbnailClick = () => {
+        setShowPlayer(true);
+        setAutoPlay(true);
+    };
+
+    const handleVideoError = (event: any) => {
+        setVideoError(true);
+        setIsLoading(false);
+        onErrorCallback(dictionary.components.videoPlayer.videoErrorText, event);
+        
+        // Clear timeout since we got a definitive error
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+    };
+    
+    const handlePlayerLoaded = () => {
+        setIsLoading(false);
+        
+        // Clear timeout since player loaded successfully
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+    };
+
+    // Error component
+    const ErrorDisplay = () => (
+        <div className="rounded-medium w-full min-w-[18rem] h-[16rem] bg-base-neutral-700 flex items-center justify-center p-4">
+            <span className="text-text-secondary text-md">
+                {dictionary.components.videoPlayer.videoErrorText}
+            </span>
+        </div>
+    );
+
+    // Loading component - shown while waiting for player to initialize
+    const LoadingDisplay = () => (
+      <div className='h-full w-full items-center justify-center flex'>
+        <div className="w-6 h-6 border-4 border-base-neutral-700 border-t-transparent rounded-full animate-spin"></div>
+        </div>
+    );
+
+    const playerStyle = videoError ? { display: 'none' } : {};
+
+    return (
+        <div className="w-full overflow-hidden">
+            {!showPlayer && thumbnailUrl ? (
+                <img
+                    src={thumbnailUrl}
+                    alt="Video Thumbnail"
+                    className="w-auto max-w-full h-auto object-contain cursor-pointer"
+                    onClick={handleThumbnailClick}
+                    onError={() => {
+                        if (!videoError) {
+                            setShowPlayer(true);
+                        }
+                    }}
+                />
+            ) : (
+                <div className="relative w-full">
+                    {/* Show error display when in error state */}
+                    {videoError && <ErrorDisplay />}
+                    
+                    {/* Show loading display when still loading and not in error state */}
+                    {isLoading && !videoError && <LoadingDisplay />}
+                    
+                    {/* Always render MuxPlayer but hide it completely with CSS when in error state */}
+                    {/* This approach allows error events to be captured without visible flickering */}
+                    <div style={playerStyle} className={isLoading && !videoError ? "opacity-0" : "opacity-100"}>
```

Comment:
The approach of using opacity to hide the MuxPlayer causes the loading spinner to not be centered. There's also a small visual bug of it appearing on top of the player for a few milliseconds.

Replacing this with `"hidden" : "block"` for display property seems to fix both these issues